### ISSUE TITLE
Middleware Fails when called from gateway

### ISSUE THREAD

**Issue by rvegas** [kusanagi/katana-sdk-node#15]
With this config:

```
<?xml version="1.0" encoding="UTF-8"?>
<middleware xmlns="urn:katana:middleware" name="api" version="1.0.0" protocol="urn:katana:protocol:http" request="true" response="true">
    <info title="Auth" description="Response middleware for API requests and responses" terms="https://tech.example.com/docs/middleware-api">
        <contact name="API" url="https://tech.example.com/docs/middleware-apii" email="api@api.com"/>
        <license name="License" url="https://tech.example.com/docs/middleware-api"/>
    </info>
    <external-docs description="API middleware docs" url="https://tech.example.com/docs/middleware-api"/>
    <engine runner="urn:katana:runner:node" path="../runner.js"/>
    <http-response code="401" description="The access credentials were invalid" mime="application/problem+json">
        <http-example>
            <![CDATA[
            {
              "type": "https://tech.example.com/api/errors#401-Unauthorized",
              "detail": "The access credentials used on this request are not valid",
              "status": 401,
              "title": "Unauthorized"
            }
            ]]>
        </http-example>
    </http-response>
</middleware>
```

And this code:
```
const Middleware = require('katana.sdk').Middleware;
const middleware = new Middleware();

middleware.request((request) => {
    return request;
});

middleware.response((response) => {
    return response;
});

middleware.run();
```

The gateway calls return this error on any type of request:
```
2017-06-09T07:58:40.104Z [ERROR] [SDK] Error running callback: response. Cannot read property 'v' of undefined TypeError: Cannot read property 'v' of undefined
at Mapper.getHttpRequest (/home/vagrant/dev/katana/middleware-api/node_modules/katana.sdk/sdk/mapper.js:101:20)
at Middleware._getResponse (/home/vagrant/dev/katana/middleware-api/node_modules/katana.sdk/sdk/component.js:234:20)
at Middleware._processCommand (/home/vagrant/dev/katana/middleware-api/node_modules/katana.sdk/sdk/middleware.js:30:33)
at Middleware._processMessage (/home/vagrant/dev/katana/middleware-api/node_modules/katana.sdk/sdk/component.js:142:10)
at emitThree (events.js:116:13)
at emit (events.js:194:7)
at Socket._flushRead (/home/vagrant/dev/katana/middleware-api/node_modules/zmq/lib/index.js:638:15)
at Socket._flushReads (/home/vagrant/dev/katana/middleware-api/node_modules/zmq/lib/index.js:676:23)
at _zmq.onReadReady (/home/vagrant/dev/katana/middleware-api/node_modules/zmq/lib/index.js:297:10)
```

If I run ```katana middleware request -c config/katana.xml -u http://test.com/1.0/auth/facebook``` it works as expected.

**Comment by rvegas** [comment]
Note, this happens on all forks, including the `async` fork with `done()` calls instead of returns.

**Comment by rvegas** [comment]
@orestes any news on this? This is quite a major blocker, I've been checking the code for the past hours and can't figure much out. What I did see from debugging is that for some reason the payload gets logged twice into the console from the mapper, the first time correctly and the second time is `undefined`. Let me know and we can pair and fix this asap.

**Comment by rvegas** [comment]
ping @orestes I have not tested this yet on latest release. but I see no mention or reference to this issue. Did you check anything regarding it?

**Comment by orestes** [comment]
Just managed to recover the vm, and this is still broken. I thought the new version would fix this but it seems unrelated. Looking into it now.

**Comment by rvegas** [comment]
thanks.

**Comment by orestes** [comment]
I'm running into a problem where it looks like the timeout is not being respected by `katana`, it timeouts right away. Getting sidetracked by this but still looking into it

**Comment by jeronimoalbi** [comment]
@orestes, we would need an example to replicate it. There are various ways of setting timeouts. I am assuming you are using the CLI options but you would have to confirm. After that I can try it with the python SDK.

**Comment by rvegas** [comment]
@jeronimoalbi @orestes any updates on this? This is causing important issues since the behaviour for middlewares is not stable.
@orestes I think this is SDK related, the following message is always shown when trying to retrieve the Service Schema from a middleware:

```
2017-07-19T18:28:44.628Z [ERROR] [SDK] Error running callback: request. Cannot get service schema. No mapping provided Error: Cannot get service schema. No mapping provided
```

I've debugged this line https://github.com/kusanagi/katana-sdk-node/blob/master/sdk/api.js#L157 and the **servicesMapping** property is always empty. Even if the services are running and registered.

**Comment by rvegas** [comment]
Fixed by #18
